### DISCUSSION TITLE
Have one folder for mixins in src

### DISCUSSION THREAD

**Discussion by confact** [luckyframework/lucky#1245] [General]
I have thought a way and done a lot of hacks to share mixins for different types, like pages, actions and operations. Having one folder that is require pretty high would solve this issue.

I have one very simple mixin module that is just cleaning up periods that is used for pages but also components and operations:
```
module Period
  def period_string
    case period
    when "14d"
      "14 days"
    when "30d"
      "30 days"
    when "60d"
      "60 days"
    when "90d"
      "90 days"
    else
      "7 days"
    end
  end
end
```
Now I have added this mixins folder, so I only need to have it on one place but before i had to duplicate it and have it on multiple places.

A similar mixin:
```
module Hashid
  def hashids
    @hashids ||= Hashids.new(salt: Lucky::Server.settings.secret_key_base, min_hash_size: HASHID_MIN_LENGTH)
  end
  def hashid
    hashids.encode([id])
  end
end
```
hashid that make random id for an integer Is used in pages and actions. I made it ugly with this one to solve this issue by adding at a method in the model to get it and using this on other places.

Last example:
```
module SubscriptionCheck
  def subscription_time?(subscription : Subscription) : Bool
    subscription.cancelled? && subscription.next_bill_at < Time.utc
  end
end
```
A check of a subscription that will be use in component, pages (to show box or not) and before pipe to redirect or not. I moved this to `src/mixins`  - Otherwise I had to duplicate code to have it on 3 places.

What do you guys think about this? I might I missed very good thing with the current.

**Comment by jwoertink** [comment]
I actually have some mixins that are shared between pages and components. Right now that mixins folder is in `components/mixins/`. I can totally see something like

```
src/
  mixins/
    actions/
    components/
    pages/
    some_global_mixin.cr
    other_global.cs
```

Then maybe Lucky just generates the `src/mixins/`, and it's up to the user to structure internally how they want?

**Reply by confact** [reply]
That was actually my thought too. That would solve the share between different types :)
